rpm/cvmfs-config-osg.spec

```diff
@@ -20,7 +20,8 @@ Provides: cvmfs-init-scripts = 1.0.22
 Obsoletes: oasis-config < 8
 Provides: oasis-config = 9
 
-Conflicts: cvmfs-config-default
+Obsoletes: cvmfs-config-default
```

Comment:
You almost always want to put a version in an Obsoletes clause, like `Obsoletes: cvmfs-config-default < 5.0`. If you don't, nobody can ever reuse the name cvmfs-config-default, because no matter what they do, this package would replace it. You might not plan to do it right now, but plans change, or someone in another repo might want to do it.

Reply:
In this case cvmfs-config-default isn't a genuinely obsolete package, because it is the package distributed by the upstream CernVM-FS project, but if somebody installs cvmfs-config-osg I always want it to always replace any previously installed cvmfs-config-default.  If somebody wants to switch back to cvmfs-config-default, they will have to first remove cvmfs-config-osg.  So I think Obsoletes is the right thing.

cvmfs-config-default is using the 1.x series of release numbers and cvmfs-config-{osg|egi} are intentionally using the 2.x series of release numbers to indicate the latter can replace the former cvmfs-config pseudo-package they all Provide, but I don't want to depend on that always being the case.

Reply:
It's not "first remove cvmfs-config-osg", it's "remove cvmfs-config-osg AND edit the OSG repos file to exclude cvmfs-config-osg so yum doesn't even see it". That's the problem with the unversioned Obsoletes.

Reply:
I see, I wasn't aware of that.  However, if I used `Obsoletes: cvmfs-config-default < 2` wouldn't it still prevent yum from seeing any current cvmfs-config-default package, which is anyway less than version 2?  It's not that much of a difference.  Since the osg repo is given a higher priority, I always have to disable it when I want to install cvmfs-config-default and the upstream cvmfs for testing, and that's probably the right thing for other people too.

Reply:
I guess my point is that having it be unversioned cuts off future possibilities.

Reply:
I'd rather cut off the possibility of a cvmfs-config-default getting to be a higher version.  Even a cvmfs-config-default version 3.x ought to be removed if a cvmfs-config-osg version 2.x is installed.

Reply:
OK. Well, I hope this won't bite us in the future.